### ISSUE TITLE
consider adding path shape type

### ISSUE THREAD

**Issue by catdad** [catdad/canvas-confetti#81]
Example code (limited to more modern browsers):

```javascript
function path(context, x, y, scaleX, scaleY, rotation, path) {
  context.save();
  context.translate(x, y);
  context.rotate(rotation);
  context.scale(scaleX, scaleY);
  context.fill(new Path2D(path));
  context.restore();
}
...
path(context, fetti.x + fetti.wobble, fetti.y + fetti.wobble, Math.abs(x2 - x1) * 0.2, Math.abs(y2 - y1) * 0.2, Math.PI / 10 * fetti.wobble, fetti.shape.path);
...
confetti({
    particleCount: 100,
    shapes: ['circle', { type: 'path', path: 'M1 1 h 8 v 8 h -8 Z' }]
});
```

A path will probably need to provide a width and height or a scale factor, so that the resulting path can be scaled appropriately.

This code has very poor performance. Any acceptable solution will need to solve that.

**Comment by catdad** [comment]
Ooh, `Path2D` can be transformed directly with a `DOMMatrix` with is far more performant than changing the canvas context, and bonus points... it takes a css transform string.

**Comment by catdad** [comment]
This has been shipped in version 1.8.0.
